src/tests/Microsoft.VisualStudio.Composition.Tests/XunitExtensions/AssertFailedException.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace Microsoft.VisualStudio.Composition.Tests
+{
+    using System;
+    using System.Runtime.Serialization;
+
+    [Serializable]
+    internal class AssertFailedException : Exception
+    {
+        public AssertFailedException()
```

Comment:
Yes: public APIs have a very real long term cost. I want to make sure I only pay for it for the individual members that need it. :)
Of course it's much less of a problem (if one at all) in test assemblies.